README.md

```diff
@@ -18,210 +18,294 @@ Darkflame Universe is licensed under AGPLv3, please read [LICENSE](LICENSE). Som
+### Building on MacOS
+In the repository root folder run the following. Ensure -DOPENSSL_ROOT_DIR=/path/to/openssl points to your openssl install location.  This is generally located in or around `/usr/local/opt/openssl@3/` but may differ.  For example, if openssl is located at `/usr/local/opt/openssl@3/` then the full cmake command would be `cmake .. -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl@3/`
 
-* All that you need to do is create a database to connect to.  As long as the server can connect to the database, the schema will always be kept up to date when you start the server.
+```bash
+# Create the build directory, preserving it if it already exists
+mkdir -p build
+cd build
 
-#### Configuration
+# Run CMake to generate build files
+cmake .. -DOPENSSL_ROOT_DIR=/path/to/openssl
 
-After the server has been built there should be four `ini` files in the build director: `sharedconfig.ini`, `authconfig.ini`, `chatconfig.ini`, `masterconfig.ini`, and `worldconfig.ini`. Go through them and fill in the database credentials and configure other settings if necessary.
+# Get cmake to build the project. If make files are being used then using make and appending `-j` and the amount of cores to utilize may be preferable, for example `make -j8`
+cmake --build . --config Release
+```
 
-#### Migrations
+### Building on **all** other platforms
+You can either run `build.sh` when in the root folder of the repository:
 
-The database is automatically setup and migrated to what it should look like for the latest commit whenever you start the server.
+```bash
+./build.sh
+```
+
+Or manually run the commands used in [build.sh](build.sh).
+
+## Updating your server
+To update your server to the latest version navigate to your cloned directory
+```bash
+cd /path/to/DarkflameServer
+```
+Run the following commands to update to the latest changes
+```bash
+git pull
+git submodule update --init --recursive
+```
+Now follow the [build](#build-the-server) section for your system and your server is up to date.
+
+## Configuring your server
+This server has a few steps that need to be taken to configure the server for your use case.
 
-#### Verify
+### Setting up the client dependencies
+Darkflame Universe can run with either a packed or an unpacked client as shown [here](#lego®-universe-11064)
+Navigate to `sharedconfig.ini` and fill in the `client_path` field with the location of your client.
 
+### Setting up the database
+Darkflame Universe utilizes a MariaDB database for account and character information.
+
+Initial setup can vary drastically based on which operating system or distribution you are running; there are instructions out there for most setups, follow those and come back here when you have a database up and running.
+
+Note that all that you need to do is create a database to connect to.  As long as the server can connect to the database, the schema will always be kept up to date when you start the server.
```

Comment:
This seems like duplicate information given that it was already specified in the dependency section